src/main/kotlin/mate/academy/model/Square.kt

```diff
@@ -0,0 +1,35 @@
+package mate.academy.model
+
+import mate.academy.service.ColorSupplier
+import kotlin.random.Random
+
+class Square(
+    color: Color,
+    val side: Int,
+) : Figure(color), BehaviorFigures {
+
+    override fun getArea() = side.times(side).toDouble()
+
+    override fun drawFigure(figure: Figure) {
+        val figureName = getFigureName(figure)
+        println(
+            """Figure: $figureName, 
+                |area: ${getArea()} sq.units, 
+                |side: $side units, 
+                |color: ${getColor(figure)}
+                |
+                """.trimMargin()
+        )
+    }
+
+    companion object {
+        private val colorSupplier = ColorSupplier()

```

Comment:
Passin to each figure class ColorSupplier it is not optimal solution.
Consider passing it as a parameter from FigureSupplier class.

If you want achieve static-style behaviour , optional you may consider using extension function